### PR TITLE
Full Locale support

### PR DESCRIPTION
**Pull request by gburgett**
This PR adds locale support to the gem.  I've been working on it for a while and finally got a spot where I think it can be merged and tested.

This PR also corrects a mistake I made when designing the gem the first time.  Currently the CDN Adapter always requests `locale=*` to get entry fields in this format:
```json
"fields" {
  "title" {
    "en-US": "foobar"
  }
}
```
It made it easier to integrate it with the Model layer, because the Sync endpoint also gets entries in that format and we store them in the sync stores in that format.  But ultimately, it's better to request the specific locales we want.  To this end I've modified most of the query fixtures to be the `locale=en-US` format, rather than the `locale=*` format.

I was able to accomplish the translation between the `locale=*` format and the `locale=en-US` format using a Store middleware, that made it really easy to isolate that code!

### CODE REVIEW

**Comment by watermarkchurch-deployment** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big pull request - please break it up into smaller units of work</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">No issue referenced - please create an issue describing a single unit of work and reference it using "closes #[the issue number]"</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: wcc-contentful/lib/wcc/contentful/model_builder.rb [hunk 1]
```diff
@@ -63,6 +63,12 @@ def build_model(typedef)
               raise ArgumentError, 'Wrong Content Type - ' \
                                    "'#{raw.dig('sys', 'id')}' is a #{ct}, expected #{typedef.content_type}"
             end
+            if raw.dig('sys', 'locale').blank?
+              raise ArgumentError, 'Model layer cannot represent "locale=*" entries. ' \
```
Comment (gburgett) [inline wcc-contentful/lib/wcc/contentful/model_builder.rb:67]: Here's an important change.  It never made sense to pass a `locale=*` entry to `Page.new`.  So now the model layer expects a localized entry and will throw if it doesn't get one.

File: wcc-contentful/lib/wcc/contentful/simple_client/cdn.rb [hunk 2]
```diff
@@ -81,13 +81,16 @@ def content_types(query = {})
   def sync(sync_token: nil, **query, &block)
     return sync_old(sync_token: sync_token, **query) unless block_given?
 
-    sync_token =
+    query = {
+      # override default locale for sync queries
+      locale: nil
```
Comment (gburgett) [inline wcc-contentful/lib/wcc/contentful/simple_client/cdn.rb:86]: Here's one case where our VCRs will change.  The URL will go from `/sync?locale=*` to just `/sync`.

File: wcc-contentful/lib/wcc/contentful/store/cdn_adapter.rb [hunk 3]
```diff
@@ -29,7 +29,7 @@ def initialize(client = nil, preview: false)
     end
 
     def find(key, hint: nil, **options)
-      options = { locale: '*' }.merge!(options || {})
+      options = options&.dup || {}
```
Comment (gburgett) [inline wcc-contentful/lib/wcc/contentful/store/cdn_adapter.rb:32]: Another case of our URLs changing.  Previously we would append `locale=*` to every query, now we don't.

File: wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb [hunk 4]
```diff
@@ -26,6 +30,8 @@
   it_behaves_like 'contentful store', {
     nested_queries: true,
     include_param: true,
+    # TODO: - fix locale queries
```
Comment (watermarkchurch-deployment) [inline wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb:33]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/62b6395716d993b0c3c236623b7ccb14e8779f31/wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb#L33"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/62b6395716d993b0c3c236623b7ccb14e8779f31/wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb">wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

**Comment by gburgett** [comment]
Used in https://github.com/watermarkchurch/reengage.com/pull/330

File: docs/1.4/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+41
+42
+43
+44</pre>
+    </td>
+    <td>
+      <pre class="code"><span class="info file"># File 'lib/wcc/contentful/store/instrumentation.rb', line 40</span>
+
+<span class='kw'>def</span> <span class='id identifier rubyid_find_all'>find_all</span><span class='lparen'>(</span><span class='op'>**</span><span class='id identifier rubyid_params'>params</span><span class='rparen'>)</span>
+  <span class='comment'># end happens when query is executed - todo.
```
Comment (watermarkchurch-deployment) [inline docs/1.4/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html:284]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/8c0858184411bd9f7dec2630a8a1382cfff5a828/docs/1.4/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html#L284"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/8c0858184411bd9f7dec2630a8a1382cfff5a828/docs/1.4/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html">docs/1.4/wcc-contentful/WCC/Contentful/Store/InstrumentationWrapper.html</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

File: wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb (hunk 4 above)
Comment (watermarkchurch-deployment) [inline wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb:33]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/watermarkchurch/wcc-contentful/blob/8c0858184411bd9f7dec2630a8a1382cfff5a828/wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb#L33"></span>TODO added in <a href="https://github.com/watermarkchurch/wcc-contentful/blob/8c0858184411bd9f7dec2630a8a1382cfff5a828/wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb">wcc-contentful/spec/wcc/contentful/store/postgres_store_spec.rb</a> - is there a card associated with that?</td>
    </tr>
  </tbody>
</table>

**Pull request merged**